lib/asset/serve-snap/src/Obelisk/Asset/Serve/Snap.hs

```diff
@@ -94,8 +95,15 @@ serveAsset' doRedirect base fallback p = do
           let finalFilename = base </> p </> "encodings" </> T.unpack (decodeUtf8 e)
           stat <- liftIO $ getFileStatus finalFilename
           modifyResponse $ setHeader "Last-Modified" "Thu, 1 Jan 1970 00:00:00 GMT"
-          modifyResponse $ setResponseCode 200 . setContentLength (fromIntegral $ fileSize stat) . setContentType (fileType defaultMimeTypes p)
-          sendFile finalFilename
+          modifyResponse $ setHeader "Accept-Ranges" "bytes"
+          modifyResponse $ setContentType (fileType defaultMimeTypes p)
+          let size = fromIntegral $ fileSize stat
+          -- Check if this is a range request
+          req <- getRequest
+          wasRange <- checkRangeReq req finalFilename size
+          unless wasRange $ do
+            modifyResponse $ setResponseCode 200 . setContentLength size
+            sendFile finalFilename
```

Comment:
What happens when we *don't* `sendfile`?

Reply:
`checkRangeReq` is a poorly named function which actually does all of the work for range requests.

Reply:
Haha I suspected as much... Maybe a comment then.